### ISSUE TITLE
NSwag generates something even when swagger.json is empty

### ISSUE THREAD

**Issue by Kikimora** [RSuter/NSwag#1968]
I think I'm running into some very stupid mistake. Basically NSwag generates ASP.NET controller full of methods but I have no idea where it finds swagger.json file to start with.

I created empty folder and sample nswag.json in the folder. There is just this file in the folder.
```json
{
    "runtime": "NetCore21",
    "codeGenerators": {
        "swaggerToCSharpController":{                        
            "useLiquidTemplates":true,
            "AspNetNamespace":"Microsoft.AspNetCore.Mvc",
            "ControllerBaseClass":"Microsoft.AspNetCore.Mvc.Controller",
            "namespace":"AA.BB",
            "UseCancellationToken":true,
            "responseArrayType":"System.Collections.Generic.List",
            "ClassStyle":"poco",
            "ControllerStyle":"abstract",
            "output":"ResourceController.cs",
            "IsAspNetCore":true
        }
    }
}
```
Then I run 
```rm ResourceController.cs; dotnet "<path to nuget>/nswag.msbuild/11.17.21/build/NetCore21/dotnet-nswag.dll" run```
And get ResourceController.cs full of methods. 

So my question boils down to - how NSwag searches for input swagger files in my scenario?

**Comment by Kikimora** [comment]
After digging into source code I figured this out, correct nswag.json is 
```json
{
    "runtime": "NetCore22",
    "swaggerGenerator":{
        "fromSwagger":{
            "json":"swagger.json"
        }
    },
    "codeGenerators": {
        "swaggerToCSharpController":{               
            "useLiquidTemplates":true,
            "AspNetNamespace":"Microsoft.AspNetCore.Mvc",
            "ControllerBaseClass":"Microsoft.AspNetCore.Mvc.Controller",
            "namespace":"AA.BB",
            "UseCancellationToken":true,
            "responseArrayType":"System.Collections.Generic.List",
            "ClassStyle":"poco",
            "ControllerStyle":"abstract",
            "output":"ResourceController.cs",
            "IsAspNetCore":true
        }
    }
}
```

Without "swaggerGenerator" NSwag picks up document from `petstore.swagger.io` by default.

**Comment by RSuter** [comment]
I think if there is no "swaggerGenerator" then the config class will just construct the default one and this means it will use the petstore sample...

**Comment by Kikimora** [comment]
This is exactly what happened :) And thank you for great tool, it saves me ton of time now. 

Confusion comes from the way input is specified. As indicated by sample, swagger2csharpcontroller command takes input from /input: command line argument. When same command invoked via nswag file it needs swaggerGenerator. 

I quickly found how to call command itself and recommendation to use nswag file. Then I discovered that there is nice correspondence between command input parameters and keys in nswag file. But then I found that this correspondence works basically for everything except input parameter.
